### PR TITLE
fix(ui): TE-2373 display all properties while creating alert in advanced mode

### PR DESCRIPTION
**Pull request by nalin-patidar**
#### Issue(s)
https://startree.atlassian.net/browse/TE-2373

#### Description
The properties that we display while creating alert in advanced mode are grouped in sections based on their step property.
However as of now, those step names are hardcoded in the UI, and because of which if there is a step that isn't there in that hardcoded list, is getting filtered out and not shown on the UI.
What we are doing with the changes of this PR is that we will be grouping the properties based on their steps dynamically, that way if there is a new category of step introduced, we wont have to make any changes and it will work and show up automatically on the UI grouped together with other properties of the same step.

#### Screenshots

Attach screenshots of UI/U

https://github.com/user-attachments/assets/1c5c3538-4d3f-49de-937c-e6f216f803a2

X changes, if applicable.

#### How to test

List steps to reproduce the issue(s) and test the change, if required.
1. Create an alert in advance mode, and see that all the properties are being displayed. You can verify the properties against the template in configuration -> alert template -> go the the specific alert template

### CODE REVIEW

File: thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx [hunk 1]
```diff
@@ -32,107 +34,125 @@ const StyledGrid = withStyles((theme) => ({
...
+                groupedProperties[ALERT_PROPERTIES_DEFAULT_STEP.STEP];
+        }
 
         return (
             <StyledGrid container>
-                {existingPropertySteps
-                    .filter((s) => !isStepEmpty(s, groupedProperties))
-                    .map((step, stepIdx) => {
-                        const subStepMap = groupedProperties[step];
+                {Object.keys(ordeeredGroupedProperties).map((step, stepIdx) => {
```
Comment (jayeshchoudhary) [inline thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx:69]: typo `ordeeredGroupedProperties` => `orderedGroupedProperties`

**Comment by jayeshchoudhary** [comment]
@nalin-patidar are the properties coming from backend?

File: thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx [hunk 2]
```diff
@@ -32,107 +34,125 @@ const StyledGrid = withStyles((theme) => ({
...
+                    ordeeredGroupedProperties[alertProperty] =
+                        groupedProperties[alertProperty];
+                }
+            }
+        });
+        const keysNotInDefaultOrder = keysPresent.filter(
+            (key) => !ALERT_PROPERTIES_DEFAULT_ORDER.includes(key)
+        );
+        if (!isEmpty(keysNotInDefaultOrder)) {
+            keysNotInDefaultOrder.forEach((key) => {
```
Comment (jayeshchoudhary) [inline thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx:58]: are we just doing this in component directly?
wont it cause a lot of re-renders?

imo we can move this to a util function and save the value in useMemo / useState

Comment (nalin-patidar) [reply thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx:58]: It wont cause any re-renders as we are creating a new object in the form we want from the incoming prop. there is no state update happening in between. Once we have the object ready we iterate over it render the component. Moving to util would be ideal if there is a repetitive use case for this kind of manipulation which we don't have as this very specific to this flow. If in future needs arises where we want to do something similar we can think and move it to some util

**Comment by nalin-patidar** [comment]
> @nalin-patidar are the properties coming from backend?

@jayeshchoudhary Yes

File: thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx (hunk 1 above)
Comment (nalin-patidar) [reply thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx]: Addressed

File: thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx [hunk 3]
```diff
@@ -32,107 +34,125 @@ const StyledGrid = withStyles((theme) => ({
...
+                                        />
+                                    )
+                                )}
+                                {Object.keys(subStepMap)
+                                    .slice(1)
+                                    .filter((cs) => subStepMap[cs].length > 0)
+                                    .map((currentSubStep) => (
+                                        <Grid
+                                            item
+                                            key={`required-${step}-${currentSubStep}`}
```
Comment (ghoshnirmalya) [inline thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx:107]: Please ensure that this is always unique.

**Review by ghoshnirmalya — APPROVED** [review]

File: thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx (hunk 3 above)
Comment (nalin-patidar) [reply thirdeye-ui/src/app/components/alert-wizard-v2/alert-template/alert-template-properties-builder/alert-template-sectioned-properties/alert-template-sectioned-properties.component.tsx:107]: They are.

**Pull request merged**